cedar-lean/UnitTest/Proto.lean

```diff
@@ -32,19 +32,50 @@ namespace Except
 instance [DecidableEq α] [DecidableEq β] : DecidableEq (Except α β) := by
   unfold DecidableEq
   intro a b
-  cases a <;> cases b <;>
-  -- Get rid of obvious cases where .ok != .err
-  try { apply isFalse ; intro h ; injection h }
-  case error.error c d =>
-    match decEq c d with
-      | isTrue h => apply isTrue (by rw [h])
-      | isFalse _ => apply isFalse (by intro h; injection h; contradiction)
-  case ok.ok c d =>
-    match decEq c d with
-      | isTrue h => apply isTrue (by rw [h])
-      | isFalse _ => apply isFalse (by intro h; injection h; contradiction)
+  cases a <;> cases b <;> simp only [reduceCtorEq, ok.injEq, error.injEq]
+  case ok.error | error.ok => exact instDecidableFalse
+  case error.error c d => exact decEq c d
+  case ok.ok c d => exact decEq c d
 end Except
 
+namespace Proto
+
+/-!
+Struct with array of UInt32 for testing purposes
+-/
+structure HardCodeStruct where
+  f6 : Array UInt32 -- Field 6
+deriving Inhabited, Repr, DecidableEq
+
+namespace HardCodeStruct
+
+def merge_6 (result : HardCodeStruct) (x : Array UInt32) : HardCodeStruct :=
+  {result with
+    f6 := (@Field.merge (Packed UInt32)) result.f6 x
+  }
+
+def merge (x : HardCodeStruct) (y : HardCodeStruct) : HardCodeStruct :=
+  {x with
+    f6 := (@Field.merge (Packed UInt32)) x.f6 y.f6
+  }
+
+def parseField (t : Tag) : BParsec (MergeFn HardCodeStruct) := do
+  match t.fieldNum with
+    | 6 =>
+      let x : Packed UInt32 ← Field.guardedParse t
+      pure (merge_6 · x)
+    | _ =>
+      t.wireType.skip
+      pure id
+
+instance : Message HardCodeStruct := {
+  parseField := parseField
+  merge := merge
+}
+
+end HardCodeStruct
+
+
 
 #guard (@Field.interpret? Bool) (ByteArray.mk #[0]) = Except.ok false
```

Comment:
Is there a reason for why we don't want to use our unit testing framework here instead of `#guard`?

Reply:
What is the advantage of our unit testing framework over `#guard` (or `theorem ... := by native_decide`)?  The latter two will fail at compile time, which seems better than waiting until the unit tests are run?  (Obviously, this doesn't work for tests that require IO, but for tests that don't require IO ...)

Reply:
Yeah, the main reason for our unit testing framework is to cover use cases like requiring IO. Another potential advantage is that the tests don't immediately fail the execution, but act more like a regular unit testing framework where results are gathered and then presented in the end. 

I don't have strong feelings here other than liking consistency :)

If we want to keep these static, maybe turn them into `theorem` so it consistent with other tests?

Reply:
> Another potential advantage is that the tests don't immediately fail the execution

Why is this an advantage?

> maybe turn them into `theorem`

Definitely can do

Reply:
> Why is this an advantage?

If you have N failing tests, and the framework fails after the first one, you won't see the next one fail after you fix the first one.  I prefer to see all the failures up front (similar to how Lean tries to continue compiling as much of your code as it can after encountering the first failure).  That's just a personal preference though.

Reply:
That's fair, and a good reason to convert

Reply:
Refactored